samcli/lib/translate/sam_template_validator.py

```diff
@@ -45,6 +45,37 @@ def __init__(self, sam_template, managed_policy_loader, profile=None, region=Non
         self.sam_parser = parser.Parser()
         self.boto3_session = Session(profile_name=profile, region_name=region)
 
+    def get_translated_template(self):
+        """
+        Runs the SAM Translator to determine if the template provided is valid and then
+        returns the translated template if it is
+
+        Returns
+        -------
+        dict
+            A dictionary representing the translated template file
+        """
+        managed_policy_map = self.managed_policy_loader.load()
+
+        sam_translator = Translator(
+            managed_policy_map=managed_policy_map,
+            sam_parser=self.sam_parser,
+            plugins=[],
+            boto_session=self.boto3_session,
+        )
+
+        self._replace_local_codeuri()
+        self._replace_local_image()
+
+        try:
+            template = sam_translator.translate(sam_template=self.sam_template, parameter_values={})
+            LOG.debug("Translated template is:\n%s", yaml_dump(template))
+            return yaml_dump(template)
+        except InvalidDocumentException as e:
+            raise InvalidSamDocumentException(
+                functools.reduce(lambda message, error: message + " " + str(error), e.causes, str(e))
+            ) from e
+
```

Comment:
this is almost all copy-paste from is_valid, should probably do some refactoring here 😅

Reply:
Removed it and combined it with is_valid. is_valid was renamed to get_translated_template_if_valid